src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2412,138 +2448,93 @@ public override BoundNode VisitConditionalOperator(BoundConditionalOperator node
                 //   object[] a = ...;
                 //   IEnumerable<object?> b = ...;
                 //   var c = true ? a : b;
-                BoundExpression consequencePlaceholder = CreatePlaceholderIfNecessary(consequence, consequenceResult);
-                BoundExpression alternativePlaceholder = CreatePlaceholderIfNecessary(alternative, alternativeResult);
+                BoundExpression consequencePlaceholder = CreatePlaceholderIfNecessary(consequence, consequenceRValue.ToTypeWithAnnotations());
```

Comment:
This comment is not correct.  Delete?

Reply:
No, I mean the *whole* comment is incorrect.  We do not base the result nested nullability on reachability.

---
In reply to: [268478296](https://github.com/dotnet/roslyn/pull/34323#discussion_r268478296) [](ancestors = 268478296)

Reply:
>  We do not base the result nested nullability on reachability.

I believe that statement agrees with the comment.